### ISSUE TITLE
Could print iter's info in benchmark?

### ISSUE THREAD

**Issue by 0400H** [huggingface/optimum-benchmark#197]
We can get summary msg after benchmark.
could print more msg in benchmark per iteration?

**Comment by IlyasMoutawwakil** [comment]
measurements from all iterations are saved in the benchmark report (in `values` field)

**Comment by 0400H** [comment]
> measurements from all iterations are saved in the benchmark report (in `values` field)

@IlyasMoutawwakil 

I use python API and only get `benchmark.log`.

Should I use cli?

**Comment by IlyasMoutawwakil** [comment]
I don't understand what you mean exactly, frim the python api, once you have `benchmark_report` you can access all values in `benchmark_report.prefill.latency.values` for example, or save the report with `benchmark_report.save_json(...)` and investigate the values https://huggingface.co/datasets/optimum-benchmark/cuda/blob/main/cuda_training_transformers_text-generation_openai-community/gpt2/benchmark_report.json#L21

**Comment by 0400H** [comment]
> I don't understand what you mean exactly.. from the python api, once you have `benchmark_report` you can access all values in `benchmark_report.prefill.latency.values` for example, or save the report with `benchmark_report.save_json(...)` and investigate the values https://huggingface.co/datasets/optimum-benchmark/cuda/blob/main/cuda_training_transformers_text-generation_openai-community/gpt2/benchmark_report.json#L21 or simply print the report object in your python environement.

Thanks for your explain.

I use the example code from [readme](https://github.com/huggingface/optimum-benchmark?tab=readme-ov-file#running-benchmarks-using-the-python-api-). This code only generate `benchmark.log` in benchmark.

There is no documentation show that how to generate `benchmark.json`、`benchmark_report.json` and so on.

Maybe i should try to use `.save_json` to generate detailed msg and read more source code.

**Comment by IlyasMoutawwakil** [comment]
Yes sorry for that we still don't have proper documentation (help is most welcome), the library is mostly for internal use and super-users of the HF ecosystem as it builds upon a lot of its principles (`push_to_hub`, `from_pretrained`, `save_pretrained`).
